mono/metadata/profiler.c

```diff
@@ -743,6 +789,38 @@ mono_profiler_class_loaded (MonoClass *klass, int result)
 	}
 }
 
+void
+mono_profiler_static_field (MonoClassField *field)
+{
+	for (ProfilerDesc *prof = prof_list; prof; prof = prof->next)
+		if ((prof->events & MONO_PROFILE_FIELD_EVENTS) && prof->static_field)
+			prof->static_field (prof->profiler, field);
+}
+
+void
+mono_profiler_static_fields_allocated (MonoClass *klass)
```

Comment:
That's a bad idea as it might force extra vtables to be allocated. mono_class_vtable is a particularly nasty function because it can throw a managed exception (yes, I know).